lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -166,10 +175,13 @@ public ColumnChunk readPrimitive(Field field)
                 readNextPage();
             }
             int valuesToRead = Math.min(remainingValueCountInPage, nextBatchSize - valueCount);
+            if (valuesToRead == 0) {
+                // When we break here, we could end up with valueCount < nextBatchSize, but that is OK.
+                break;
```

Comment:
This break is what is causing the IllegalStateException to occur in LazyBlock, @martint. How should this be handled? Should we collect info about the values that are skipped inside of BlockBuilder and then take that into account in LazyBlock?